src/common/readout/vmm3/VMM3Config.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2021 European Spallation Source, ERIC. See LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Get detector configuration from json file
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <common/JsonFile.h>
+#include <common/debug/Trace.h>
+#include <common/readout/vmm3/Hybrid.h>
+#include <string>
+#include <vector>
+
+// #undef TRC_LEVEL
+// #define TRC_LEVEL TRC_L_DEB
+
+class VMM3Config {
+public:
+  // Provide maximum values from all VMM3 detectors
+  static constexpr uint8_t MaxRing{
+      10}; // 12 (logical) rings from 0 to 11, 11 reserved for monitors
+  static constexpr uint8_t MaxFEN{13};    // This is topology specific
+  static constexpr uint8_t MaxHybrid{2}; // Hybrids are VMM >> 1
+
+  VMM3Config(){};
+
+  // Load and apply the json config
+  VMM3Config(std::string Instrument, std::string ConfigFile)
+      : ExpectedName(Instrument), FileName(ConfigFile) {}
+
+  // load file into json object and apply
+  void loadAndApplyConfig();
+
+  void loadAndApplyCalibration(std::string CalibFile);
+
+  void applyVMM3Calibration(ESSReadout::Hybrid &Hybrid, unsigned vmmid,
+                            nlohmann::json VMMCalibration);
+
+  // Apply the loaded json file
+  virtual void applyConfig() = 0;
+
+  // Get Hybrid from the Ring, FEN, and VMM numbers
+  // Currently Hybrids are stored as a 3D array, but may be updated in future
+  ESSReadout::Hybrid &getHybrid(uint8_t Ring, uint8_t FEN, uint8_t VMM) {
+    return Hybrids[Ring][FEN][VMM];
+  }
+
```

Comment:
Consider renaming getHybrid to getHybridByNumbers, getHybridByString, maybe not necessary

Reply:
One reason for the different names is to highlight differences in speed, but if getHybrid(string id) needs to be faster in future, the structure of hybrids can be re-jigged to a mapping of strings to Hybrids, for example.

Will keep names the same for now